src/adapters/MeetingsJSONAdapter.js

```diff
@@ -132,42 +132,63 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
   }
 
   /**
-   * Returns an observable that emits the display data of a meeting control.
+   * Returns a MediaStream object obtained from local user media.
+   * @param {object} meeting  ID of the meeting to get.
+   * @returns {MediaStream}
+   */
+  async getRemoteVideo(meeting) {
+    let mediaStream;
+
+    // Handle media streams changes to ready state
+    meeting.on('media:ready', (media) => {
+      if (!media) {
+        return;
+      }
+
+      if (media.type === 'remoteVideo') {
+        mediaStream = media.stream;
+      }
+      if (media.type === 'remoteAudio') {
+        mediaStream = media.stream;
+      }
+    });
+
+    return mediaStream;
+  }
+
+  /**
+   * Returns an observable that emits a Meeting object.
+   * Whenever there is an update to the meeting, the observable
+   * will emit a new updated Meeting object, if datasource permits.
    *
    * @param {string} ID ID of the meeting for which to update display
    * @returns {Observable.<MeetingControlDisplay>}
    * @memberof MeetingJSONAdapter
    * @private
    */
-  muteAudioControl(ID) {
-    const unmuted = {
-      ID: MUTE_AUDIO_CONTROL,
-      icon: 'microphone',
-      tooltip: 'Mute',
-      state: MeetingControlState.INACTIVE,
-    };
-    const muted = {
-      ID: MUTE_AUDIO_CONTROL,
-      icon: 'microphone-muted',
-      tooltip: 'Unmute',
-      state: MeetingControlState.ACTIVE,
-    };
-
-    const default$ = Observable.create((observer) => {
-      const meeting = this.datasource[ID];
-
-      if (meeting) {
-        observer.next(meeting.localAudio.muted ? muted : unmuted);
+  getMeeting(ID) {
+    const getMeeting$ = Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        console.log(observer);
+        observer.next(this.datasource[ID]);
       } else {
         observer.error(new Error(`Could not find meeting with ID "${ID}"`));
       }
 
       observer.complete();
     });
 
-    const muteEvent$ = fromEvent(document, MUTE_AUDIO_CONTROL).pipe(
-      filter((event) => event.detail.ID === ID),
-      map((event) => (event.detail.localAudio.muted ? muted : unmuted))
+    return getMeeting$.pipe(
+      flatMap((meeting) => {
```

Comment:
I'm not sure this works, since `flatMap` needs to return a value to emit next, right now, either observable is being returned